docs/user_manual/dataset-terminology.md

```diff
@@ -10,23 +10,101 @@ Some terms regarding the data structures are explained here, including the defin
+
+- **{py:class}`Indexpointer <power_grid_model.data_types.Indexpointer>`:** A 1D numpy array of int64 type used to specify sparse batches. It indicates the range of components within a scenario. For example, an indptr of [0, 1, 3, 3] indicates 4 batches with element indexed with 0 in 1st batch, [1, 2, 3] in 2nd batch and no elements in 3rd batch.
+- **{py:class}`SingleColumn <power_grid_model.data_types.SingleColumn>`:** A 1D/2D numpy array of values corresponding to a specific attribute.
+- **{py:class}`BatchColumn <power_grid_model.data_types.BatchColumn>`:** A 2D/3D numpy array of values corresponding to a specific attribute.
+
+### Dimensions of numpy arrays
+
+The dimensions of numpy arrays and the interpretation of each dimension is as follows.
+
+| **Data Type**            | **1D**                                                                                       | **2D**                                                                                                           | **3D**                                                                                                           |
+|--------------------------|----------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------|
+| **SingleArray**          | Corresponds to a single dataset.                                                             | ❌                                                                                                               | ❌                                                                                                               |
+| **DenseBatchArray**      | ❌                                                                                            | Batch number $\times$ Component within that batch                                                                       | ❌                                                                                                               |
+| **SingleColumn**   | Component within that batch.                                                                 | Component within that batch $\times$ Phases &#10024;                                                                            | ❌                                                                                                               |
+| **BatchColumn**    | ❌                                                                                            | Batch number $\times$ Component within that batch                                                                       | Batch number $\times$ Component within that batch $\times$ Phases &#10024;                                                            |
```

Comment:
please reformat

Reply:
VSCode used to give squigly lines when the table was incorrectly formatted for me. It did not give me here. I tried to format using markdownlint too. Didnt change anything.
Do you know how to make the vertical lines align?

Reply:
oh weird, i have a linter that is able to do that. do you think maybe it's caused by the unicode characters instead of raw bytes?

Reply:
Well, changing to unicode certainly helped align